### PR TITLE
Use lazy tasks API

### PR DESCRIPTION
**Pull request by eygraber**

### CODE REVIEW

**Comment by eygraber** [comment]
Full disclosure, I don't have a mac so I can't actually test it.

**Comment by russhwolf** [comment]
Thanks for this. Seems to work, and we should be better about lazy config.

Can you update it to a more recent main branch? I want it to pick up the new integration tests and make sure it all still passes.

File: kmmbridge/src/main/kotlin/ProjectExtensions.kt [hunk 1]
```diff
@@ -89,15 +90,19 @@ internal fun writeGitTagVersion(project: Project, versionString: String) {
...
     val taskWithName = "assemble${name.capitalize()}${buildTypeString}XCFramework"
-    return try {
-        tasks.findByName(taskWithName) ?: tasks.findByName(taskWithoutName)!!
-    } catch (e: NullPointerException) {
-        throw UnknownTaskException("Cannot find XCFramework assemble task. Tried ${taskWithName} and ${taskWithoutName}.", e)
+    return runCatching {
+        tasks.named(taskWithName)
+    }.recoverCatching {
+        tasks.named(taskWithoutName)
+    }.getOrElse {
```
Comment (russhwolf) [inline kmmbridge/src/main/kotlin/ProjectExtensions.kt:103]: Was there a particular reason to convert try/catch to runCatching here?

Comment (eygraber) [reply kmmbridge/src/main/kotlin/ProjectExtensions.kt:103]: I thought this looked better. I can use try catch if you want.

**Comment by eygraber** [comment]
@russhwolf rebased

**Pull request merged**